### PR TITLE
[SKOOP-108] Join a closed community

### PR DESCRIPTION
**Pull request by svetivanova**
use POST /communities/{communityId}/user-registrations to create a request to join a closed community
tests
@LebedkoDmitry could you please take a look?

### CODE REVIEW

File: src/app/communities/communities.component.spec.ts [hunk 1]
```diff
@@ -62,6 +64,32 @@ const communities: CommunityResponse[] = [
...
+  lastName: authenticatedUser.lastName,
+  email: authenticatedUser.email,
+  coach: false,
+};
+
+const communityUserRegistrations: CommunityUserRegistrationResponse[] = [
+  {
+    user: currentUser,
+    approvedByUser: true,
+    approvedByCommunity: true
```
Comment (LebedkoDmitry) [inline src/app/communities/communities.component.spec.ts]: Isn't it better to set `approvedByCommunity` to `null` here? The authenticated user sends a request to join closed community and that means the request is not approved by community yet.

File: src/app/community-view/community-view.component.spec.ts [hunk 2]
```diff
@@ -59,6 +61,23 @@ const community: CommunityResponse = {
...
+  lastName: authenticatedUser.lastName,
+  email: authenticatedUser.email,
+  coach: false,
+};
+
+const communityUserRegistrations: CommunityUserRegistrationResponse[] = [
+  {
+    user: currentUser,
+    approvedByUser: true,
+    approvedByCommunity: true
```
Comment (LebedkoDmitry) [inline src/app/community-view/community-view.component.spec.ts]: Isn't it better to set `approvedByCommunity` to `null` here? The authenticated user sends a request to join closed community and that means the request is not approved by the community yet.

File: src/app/recommended-communities/recommended-communities.component.spec.ts [hunk 3]
```diff
@@ -57,6 +61,32 @@ const communities: CommunityResponse[] = [
...
+  lastName: authenticatedUser.lastName,
+  email: authenticatedUser.email,
+  coach: false,
+};
+
+const communityUserRegistrations: CommunityUserRegistrationResponse[] = [
+  {
+    user: currentUser,
+    approvedByUser: true,
+    approvedByCommunity: true
```
Comment (LebedkoDmitry) [inline src/app/recommended-communities/recommended-communities.component.spec.ts]: Isn't it better to set `approvedByCommunity` to `null` here? The authenticated user sends a request to join closed community and that means the request is not approved by the community yet.

File: src/app/recommended-communities/recommended-communities.component.spec.ts [hunk 4]
```diff
@@ -113,6 +155,15 @@ describe('RecommendedCommunitiesComponent', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should remove the community form the list of recommended communities when user has joined to it', fakeAsync(() => {
```
Comment (LebedkoDmitry) [inline src/app/recommended-communities/recommended-communities.component.spec.ts]: I think there is a misprint in the word `form`.

File: src/app/recommended-communities/recommended-communities.component.ts [hunk 5]
```diff
@@ -16,36 +16,70 @@ import { CommunityUserResponse } from '../communities/community-user-response';
 })
 export class RecommendedCommunitiesComponent implements OnInit {
 
-  communities$: Observable<CommunityResponse[]> = of([]);
+  communities: CommunityResponse[] = [];
```
Comment (LebedkoDmitry) [inline src/app/recommended-communities/recommended-communities.component.ts]: I think I got the reason why you turned `Observable` into a simple array. And this is a necessity to filter joined community out. But could we stay with `Observable` and filter it with https://www.learnrxjs.io/operators/filtering/filter.html ? After a user joined a community we can replace the original `Observable` with a filtered one. And when the component is refreshed the new list of recommended communities without a joined one will be provided by the server. What do you think about it?

File: src/app/communities/communities.component.spec.ts (hunk 1 above)
Comment (svetivanova) [reply src/app/communities/communities.component.spec.ts]: thank you. done

File: src/app/community-view/community-view.component.spec.ts (hunk 2 above)
Comment (svetivanova) [reply src/app/community-view/community-view.component.spec.ts]: thank you. done

File: src/app/recommended-communities/recommended-communities.component.spec.ts (hunk 3 above)
Comment (svetivanova) [reply src/app/recommended-communities/recommended-communities.component.spec.ts]: thank you. done

File: src/app/recommended-communities/recommended-communities.component.spec.ts (hunk 4 above)
Comment (svetivanova) [reply src/app/recommended-communities/recommended-communities.component.spec.ts]: thank you. done

**Pull request merged**